### ISSUE TITLE
 Intermittent test failures when calling runWithContext after upgrade to Scio 0.11.2

### ISSUE THREAD

**Issue by chrisheaththomas** [spotify/scio#4206]
Intermittent test failures when calling `runWithContext` after upgrade to Scio 0.11.2:

```
java.lang.IllegalArgumentException: PipelineOptions specified failed to serialize to JSON.
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:171)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:67)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:323)
	at org.apache.beam.sdk.testing.TestPipeline.run(TestPipeline.java:398)
	at org.apache.beam.sdk.testing.TestPipeline.run(TestPipeline.java:334)
    <snip>
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Unexpected IOException (of type java.io.IOException): Failed to serialize and deserialize property 'providerRuntimeValues' with value '{}'
	at com.fasterxml.jackson.databind.JsonMappingException.fromUnexpectedIOE(JsonMappingException.java:334)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsBytes(ObjectMapper.java:3769)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:168)
	... 75 more
```

Only observed with parallel test execution. Running 100 of the same test in parallel gives 0-3 failures. Running 1000 of the same test in serial consistently gives 0 failures.

Not seeing this on a 0.11.0 repo.

**Comment by kyprifog** [comment]
+1 I just got this one

**Comment by chrisheaththomas** [comment]
PR opened against beam to fix: https://github.com/apache/beam/pull/16680

**Comment by chrisheaththomas** [comment]
Possibly fixed by #4281

**Comment by kellen** [comment]
This will be fixed by #4318 ... 2.37 had an issue with joins so we're skipping it.

**Comment by chrisheaththomas** [comment]
#4318 merged.
